lambdas/nino-check/src/handler.ts

```diff
@@ -0,0 +1,175 @@
+import { Context } from "aws-lambda";
+import { randomUUID } from "crypto";
+import { getTokenFromOtg } from "./utils/otg-api";
+import { LogHelper } from "../../logging/log-helper";
+import { OtgTokenResponse } from "./types/otg";
+import { MetricsHelper } from "../../logging/metrics-helper";
+import { getRecordBySessionId } from "../../common/src/database/get-record-by-session-id";
+import { updateRecordBySessionId } from "../../common/src/database/update-record-by-session-id";
+import { insertRecord } from "../../common/src/database/insert-record";
+import { initOpenTelemetry } from "../../open-telemetry/src/otel-setup";
+import { PersonIdentityItem } from "../../common/src/database/types/person-identity";
+import { PdvApiResponse } from "./types/pdv";
+import { matchUserDetailsWithPdv } from "./utils/pdv-api";
+import { ISO8601DateString, UnixTimestamp } from "./types/brands";
+import { toEpochSecondsFromNow } from "./utils/time/date-time";
+import { TimeUnits } from "./utils/time/time-units";
+import { NinoSessionItem } from "./types/nino-session-item";
+import { AttemptItem } from "./types/attempt";
+
+initOpenTelemetry();
+
+type InputEvent = {
+  otgApiUrl: string;
+  pdvApiUrl: string;
+  pdvUserAgent: string;
+  govJourneyId: string;
+  sessionTable: string;
+  personIdentityTable: string;
+  attemptTable: string;
+  ninoUserTable: string;
+  sessionId: string;
+  nino: string;
+};
+
+const metricsHelper = new MetricsHelper();
+
+export async function handler(
+  {
+    otgApiUrl,
+    pdvApiUrl,
+    pdvUserAgent,
+    govJourneyId,
+    sessionTable,
+    personIdentityTable,
+    attemptTable,
+    ninoUserTable,
+    sessionId,
+    nino,
+  }: InputEvent,
+  context: Context
+): Promise<void> {
+  const handlerStartTime = new Date().toISOString() as ISO8601DateString;
+
+  const logHelper = new LogHelper(context);
+  const { logger } = logHelper;
+
+  logHelper.logEntry(context.functionName, govJourneyId);
+
+  const sessions = await getRecordBySessionId<NinoSessionItem>(
+    sessionTable,
+    sessionId,
+    logHelper.logger
+  );
+  logger.info(
+    `Retrieved ${sessions.length} session(s) for sessionId ${sessionId}.`
+  );
+
+  // Get the latest session, if multiple are present
+  // NB: state machine as currently implemented just returns sessions[0]. Should I switch to that?
+  const session = sessions.reduce((acc, cur) =>
+    acc.expiryDate > cur.expiryDate ? acc : cur
+  );
+
+  const personIdentities = await getRecordBySessionId<PersonIdentityItem>(
+    personIdentityTable,
+    sessionId,
+    logHelper.logger
+  );
+  logger.info(
+    `Retrieved ${personIdentities.length} person identit[y/ies] for sessionId ${sessionId}.`
+  );
+
+  // Get the latest session, if multiple are present
+  // NB: state machine as currently implemented just returns personIdentities[0]. Should I switch to that?
```

Comment:
Yeah it's because we do a `Query` operation as it's safer than a `GetItem` - primarily because GetItem throws when there's no record but with a query we can gracefully handle no record found without an exception. By doing it this way, the Query will always return an array even though, we are effectively doing Get which would have returned 1 record and not an array. So [0] should be OK imo

Reply:
@surajkumar - we handle the zero-row response inside the function we write last week I think - I'm really asking about the case where multiple records get returned here. In that case, is the first entry enough, or should we do something like I've done here where I pick the one with the latest expiry date?

Reply:
There should only ever be 1 entry. So picking the first is OK. `sessionId` is a hash: https://github.com/govuk-one-login/ipv-cri-common-lambdas/blob/f6766fec9a26a5b6f68562f75dde7a8329185f4a/infrastructure/lambda/template.yaml#L1345

Reply:
@surajkumar ah awesome, so it's always gonna be unique. Great, I'll just use `[0]` then. Cheers :)